dotCMS/src/main/java/com/dotcms/api/system/event/SystemEventsAPI.java

```diff
@@ -39,15 +39,23 @@
 	 * Pushes a new System Event to the message queue. The {@link SystemEvent}
 	 * is supposed to contain all the information it needs.
 	 * 
-	 * @param event
+	 * @param systemEvent
 	 *            - The {@link SystemEvent} object.
 	 * @throws DotDataException
 	 *             An error occurred when saving the event.
 	 */
 	public void push(SystemEvent systemEvent) throws DotDataException;
 
+	/**
+	 * Pushg an event with a payload.
+	 * @param event {@link SystemEventType}
+	 * @param payload {@link Payload}
+	 * @throws DotDataException
+     */
 	public void push(SystemEventType event, Payload payload) throws DotDataException;
 
+	// todo:
```

Comment:
oh I was going to document the async method to push events in a separated thread, but then I realized the issue is the calculation of the payload rather than just saving it.